projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -324,69 +360,35 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
   public onFocus(): void {
     this._isFocused = true;
     this.onTouchCallback();
-    this.updateMask();
+    if (this.value) {
```

Comment:
after updating `updateMask` method this check is not necessary, neither the else clause. Just call `updateMask`